hack/proxy-infrastructure/setup.go

```diff
@@ -228,23 +228,19 @@ func generate(ctx context.Context, conf *Config) (map[string]interface{}, error)
 		return nil, err
 	}
 
-	resources := []interface{}{
-		vnet(conf),
-		ip(conf),
-		nsg(conf),
-		nic(conf),
-		vm(conf),
-		cs,
+	resources := []arm.Resource{
```

Comment:
`[]*arm.Resource` ? (occurs multiple times in PR).  Let's avoid the shallow copying implied by all this dereferencing.

Reply:
Makes sense. I think, it will require to change `arm.Template` definition, but it should not be a problem.